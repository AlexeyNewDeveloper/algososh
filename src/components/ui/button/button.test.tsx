import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Корректная отрисовка кнопки", () => {
  it("Отрисовка кнопки с текстом", () => {
    const tree = renderer.create(<Button text="Кнопка с текстом" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Отрисовка кнопки без текста", () => {
    const tree = renderer.create(<Button text="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Отрисовка заблокированной кнопки", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Отрисовка кнопки с индикацией загрузки", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Корректный вызов коллбека на кнопке", () => {
    const mockCallback = jest.fn();

    render(<Button text="Кнопка" onClick={mockCallback} />);
    const button = screen.getByText("Кнопка");
    fireEvent.click(button);

    expect(mockCallback).toHaveBeenCalled();
  });
});
