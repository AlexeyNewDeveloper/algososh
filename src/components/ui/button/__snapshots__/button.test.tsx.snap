// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Корректная отрисовка кнопки Отрисовка заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Корректная отрисовка кнопки Отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Корректная отрисовка кнопки Отрисовка кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Корректная отрисовка кнопки Отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка с текстом
  </p>
</button>
`;
